connector/README.md

```diff
@@ -0,0 +1,220 @@
+# Connectors
```

Comment:
I'd expect this to live in the core repo, as with all other component types

Reply:
I agree, except in the doc I am referring to components I would expect to exist in in contrib, which seems circular. I'm happy to move it over there if don't think that's a problem.

Reply:
I've made the PR on core: https://github.com/open-telemetry/opentelemetry-collector/pull/7035

If we merge that first I'll update this PR to reference it.